xayn-ai/src/ltr/list_net/ndutils/mod.rs

```diff
@@ -258,4 +312,42 @@ mod tests {
 
         assert_approx_eq!(f32, cost, 4.300_221_4);
     }
+
+    #[test]
+    fn test_he_normal_weight_init() {
+        let mut rng = rand::thread_rng();
+        let weights = he_normal_weights_init(&mut rng, 300, 200);
+
+        assert_eq!(weights.shape(), &[300, 200]);
+
+        let std = SQRT_2 / (weights.shape()[0] as f32).sqrt();
+        let limit = 2. * std;
+        let mut c_1std = 0;
+        let mut c_2std = 0;
+        for &w in weights.iter() {
+            assert!(
+                -limit <= w && w <= limit,
+                "out of bound weight: {} <= {} <= {}",
+                -limit,
+                w,
+                limit
+            );
+            if -std <= w && w <= std {
+                c_1std += 1;
+            } else {
+                c_2std += 1;
+            }
+        }
+
+        let nr_weights = weights.len() as f32;
+        let prob_1std = c_1std as f32 / nr_weights;
+        let prob_2std = c_2std as f32 / nr_weights;
+
+        // Probabilities of a weight being in +-1std or +-2std corrected
+        // wrt. the truncating of values outside of +-2std. We accept this
+        // to be true if the found percentage is around +-5% of the expected
+        // percentage.
+        assert_approx_eq!(f32, prob_1std, 0.713_389_1, epsilon = 0.05);
+        assert_approx_eq!(f32, prob_2std, 0.284_518_84, epsilon = 0.05);
```

Comment:
it shouldn't make much of a difference, but i get slightly different probablilites ([playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=fa30ae9782fec47cf21e63c3d746f8e6)):
```suggestion
        assert_approx_eq!(f32, prob_1std, 0.715_232_8, epsilon = 0.05);
        assert_approx_eq!(f32, prob_2std, 0.284_767_2, epsilon = 0.05);
```

Reply:
I did use much less precise numbers for the calculation and calculated it using ipython and then converted it to f32 (as the difference in precision shouldn't matter given the sample size and a epsilon of 0.05).

I fixed it to use your numbers.